netbox_inventory/forms/models.py

```diff
@@ -223,10 +237,28 @@ class Meta:
 
 
 class DeliveryForm(NetBoxModelForm):
+    contact_group = DynamicModelChoiceField(
+        queryset=ContactGroup.objects.all(),
+        required=False,
+        null_option='None',
+        label='Contact Group',
+        help_text='Filter receiving contacts by group',
+        initial_params={
+            'contacts': '$receiving_contact',
+        }
+    )
+    receiving_contact = DynamicModelChoiceField(
+        queryset=Contact.objects.all(),
+        help_text=Delivery._meta.get_field('receiving_contact').help_text,
+        query_params={
+            'group_id': '$contact_group',
+        },
+    )
```

Comment:
receiving_contact form field should have `required=False` since Delivery model receiving_contact field is optional.